code/modules/mob/mob_helpers.dm

```diff
@@ -22,6 +22,40 @@
 			return 0
 	return 1
 
+/mob/proc/get_screen_colour()
+
+/mob/dead/observer/get_screen_colour()
+	return MATRIX_DEFAULT
+
+/mob/proc/update_colour(var/time = 50, var/forceupdate = 0) //Update the mob's client.color if the colour matrix in the eyes (or whatever) is different.
+	if(!client || (client.updating_colour && !forceupdate))
+		return
+	var/list/colour_to_apply = get_screen_colour()
+	var/list/difference = difflist(client.color, colour_to_apply)
+	if(difference || !(client.color) || !istype(difference) || !difference.len)
+		client.updating_colour = 1
+		if(forceupdate)
+			time = 0
+		client.colour_transition(colour_to_apply, time = time)

```

Comment:
Also, this proc is called `update_colour`, but should really be `update_client_color`. Er... `update_client_colour`. I'd really prefer if we didn't use "colour", though, since it's not consistent with BYOND's own spelling. :eyes: